### ISSUE TITLE
Re-deploy Valhalla Website & Maintain Discord Channel Backups

### ISSUE THREAD

**Issue by ECWireless** [raid-guild/RIPs#189]
# Project Idea

### Project Submitted By

@ECWireless 

## Summary

Since @saimano is no longer maintaining the Valhalla website, it is required that someone else re-deploy the site and own the continuous updating of it. For the website: the link to the original Valhalla repo is [here](https://github.com/manolingam/the-valhalla), and all that's really required is for this to be re-deployed to RaidGuild's Vercel. Reach out to @ECWireless for env file variables.

The more important task is the regular updating of Valhalla in order to make sure the site is in sync with the channels that have been sent to Valhalla from the Discord. This is most likely relatively straightforward, but it is currently a manual task. This RIP will allow for 2 total outcomes:
1. Rewarding someone in cookies for re-deploying the website to Vercel, plus someone for manually syncing the site with Discord. This RIP will also propose a regular cookie reward for whoever does this on a quarterly basis
2. Rewarding someone for designing and implementing a more automated way to backup Valhalla channels, then with proper checks, delete the backed up channels that are in Discord
    - Bonus if you can train (securely) an LLM model, so that we can quickly write case-studies and retrospectives

## Why should we build this?

The Valhalla webiste most likely hasn't been updated in a year now, and for the guild to maintain proper record-keeping, it's crucial that this gets up and running again.

## Anything else you'd like to add?

Unfortunately, this may not be a job for a non-apprentice or non-member, since it deals with sensitive information. This will be left up to a guild-wide decision in a future meeting

### For Outcome 1

- Frontend
- Anyone willing to backup channels on a quarterly basis

### For Outcome 2

- Frontend
- Backend

## Cost (in USD)

_To be paid out in cookies._

### Outcome 1

- Website deployment: $200
- Quarterly Valhalla syncing with Discord: $200 per quarter

### Outcome 2

- TBD by anyone wanting to step up for this

**Comment by ECWireless** [comment]
A note for outcome 2: we should try to design it in a way that is easily deploy-able and generic enough to be productized

**Comment by Fluffy9** [comment]
Am interested. How do I sign up? (Pupcakes)

**Comment by ECWireless** [comment]
> Am interested. How do I sign up? (Pupcakes)

Are you interested in Outcome 1 or 2? @wtfsayo is also potentially jumping on both of these, so would be good to coordinate on here who wants to do what. Should note too that Outcome 1 is definitely just a 1 person job, and should only take 1-2 hours to do.

**Comment by Fluffy9** [comment]
Outcome 2

**Comment by wtfsayo** [comment]
@ECWireless do you know how @manolingam previously exported server chats manually?

**Comment by Fluffy9** [comment]
@wtfsayo did you finish all of part 1? If so does that mean it's fine if part 2 is done a bit slower but before next quarter @ECWireless? I have a paper I should be writing as well but would like to do it if it's not time sensitive lol

**Comment by wtfsayo** [comment]
can be slow, yes Part 1 is done

**Comment by ECWireless** [comment]
> @wtfsayo did you finish all of part 1? If so does that mean it's fine if part 2 is done a bit slower but before next quarter @ECWireless? I have a paper I should be writing as well but would like to do it if it's not time sensitive lol

Yeah, part 2 can definitely be slower. Gonna take a bit of designing anyway... it's really up to how fast y'all want to work. The only risk is someone else scooping it up if it starts dragging.

**Comment by ECWireless** [comment]
> @ECWireless do you know how @manolingam previously exported server chats manually?

Linking here for posterity: https://github.com/Tyrrrz/DiscordChatExporter

**Comment by Fluffy9** [comment]
Cool. I'm free to work on this now. @wtfsayo are you intending on joining? How much money is allocated for this?

**Comment by ECWireless** [comment]
> Cool. I'm free to work on this now. @wtfsayo are you intending on joining? How much money is allocated for this?

Y'all can come up with your own proposed cost, once you've come up with a design

**Comment by Fluffy9** [comment]
![PupcakesValhallaMaintenanceProposal](https://github.com/user-attachments/assets/3993ba12-cf8f-483a-9691-4d38910dc463)
(Akash network also has a hackathon going on soon with 45k in prizes)

**Comment by wtfsayo** [comment]
> ![PupcakesValhallaMaintenanceProposal](https://github.com/user-attachments/assets/3993ba12-cf8f-483a-9691-4d38910dc463)
> 
> (Akash network also has a hackathon going on soon with 45k in prizes)
> 
> 

I think we should loop in @TravisWyche for what he has in store as he is exploring building content pipeline system

**Comment by Fluffy9** [comment]
Updates on this here: https://hackmd.io/xCtk9uMGTUWRhSBAWIbkAg

Last things needed is help to get bot permissions to work with RaidGuild server, and get it hosted

**Comment by ECWireless** [comment]
- Outcome 1 is complete; going to close this since Outcome 2 has move to #208
